site/hooks/useWeb3/web3context.tsx

```diff
@@ -0,0 +1,29 @@
+import React, { ReactChild, createContext, useContext } from "react";
+
+import { useWeb3 } from ".";
+import { Web3ProviderState, web3InitialState } from "./reducers";
+
+const Web3Context = createContext<Web3ProviderState>(web3InitialState);
+
+interface Props {
+  children: ReactChild;
+}
+
+export const Web3ContextProvider = ({ children }: Props) => {
```

Comment:
for future reference I think this is what the react FC type is for:
```ts
// props.children is typed automatically
export const Web3ContextProvider: FC = (props) => {
```

Reply:
Thanks, that was a bit of copy pasta 😛